crates/polars-ops/src/series/ops/is_first.rs

```diff
@@ -37,11 +36,53 @@ fn is_first_bin(ca: &BinaryChunked) -> BooleanChunked {
 fn is_first_boolean(ca: &BooleanChunked) -> BooleanChunked {
     let mut out = MutableBitmap::with_capacity(ca.len());
     out.extend_constant(ca.len(), false);
-    if let Some(index) = arg_max_bool(ca) {
-        out.set(index, true)
-    }
-    if let Some(index) = ca.first_non_null() {
-        out.set(index, true)
+
+    if ca.null_count() == ca.len() {
+        out.set(0, true);
+    } else if ca.null_count() == 0 && ca.chunks().len() == 1 {
+        // fast path
+        let arr = ca.downcast_iter().next().unwrap();
+        let mask = arr.values();
+        out.set(first_set_bit(mask), true);
+        out.set(first_unset_bit(mask), true);
+    } else {
+        let mut first_true_found = false;
+        let mut first_false_found = false;
+        let mut first_null_found = false;
+        let mut all_found = false;
+        ca.into_iter().enumerate().find_map(|(idx, val)| match val {
```

Comment:
Can we use `apply` here and a `index` incrementer outside of the closure? That should iterate faster and I believe that should be possible here (not tested).

Reply:
Emm, The reason for choosing `into_iter + find_map` is that we only need to find the first occurrence value(`true, false and None`), but `map` is not short-circuits. However the overhead introduced by over the entire chunkedArray could be even larger than walking through all elements(depends on the size of the array). 🤔

Reply:
Thank you for the detailed examples, which are very helpful. I updated the implementation of `is_first` to this pattern. But as for `is_last`, we need to divide each `u64` in reverse order. Do we have a util function similar to `Bitmap.chunks::<u64>` but for the reverse order? If not, do we need to introduce it to upstream or just keep it as-is.

Reply:
No, we would need to write something like that ourselves. Ideally in polars-arrow.